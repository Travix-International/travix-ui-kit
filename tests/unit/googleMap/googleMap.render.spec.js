import { shallow } from 'enzyme';
import React from 'react';
import google from './googleApiObjectMock';
import GoogleMap from '../../../components/googleMap/googleMap';

describe('GoogleMap: render', () => {
  it('should return base GoogleMap container', () => {
    const component = shallow(
      <GoogleMap google={google} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return base GoogleMap container with custom size', () => {
    const component = shallow(
      <GoogleMap google={google} height="150px" width="250px" />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return base GoogleMap container with custom marker', () => {
    const points = [{
      position: { lat: 52.379189, lng: 4.899431 },
      customMarker: true,
      info: {
        title: 'title',
      },
    }];

    const component = shallow(
      <GoogleMap
        google={google}
        height="150px"
        points={points}
        width="250px"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render GoogleMap container with provided className', () => {
    const wrapper = shallow(
      <GoogleMap className="test-class" google={google} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render GoogleMap container with provided dataAttrs', () => {
    const wrapper = shallow(
      <GoogleMap dataAttrs={{ 'test-data-attr': 'test' }} google={google} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
