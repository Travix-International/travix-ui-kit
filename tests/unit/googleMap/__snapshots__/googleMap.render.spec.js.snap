// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap: render should render GoogleMap container with provided className 1`] = `
<div
  className="test-class ui-google-map"
  style={
    Object {
      "height": "400px",
      "width": "400px",
    }
  }
/>
`;

exports[`GoogleMap: render should render GoogleMap container with provided dataAttrs 1`] = `
<div
  className="ui-google-map"
  data-test-data-attr="test"
  style={
    Object {
      "height": "400px",
      "width": "400px",
    }
  }
/>
`;

exports[`GoogleMap: render should return base GoogleMap container 1`] = `
<div
  className="ui-google-map"
  style={
    Object {
      "height": "400px",
      "width": "400px",
    }
  }
/>
`;

exports[`GoogleMap: render should return base GoogleMap container with custom marker 1`] = `
<div
  className="ui-google-map"
  style={
    Object {
      "height": "150px",
      "width": "250px",
    }
  }
/>
`;

exports[`GoogleMap: render should return base GoogleMap container with custom size 1`] = `
<div
  className="ui-google-map"
  style={
    Object {
      "height": "150px",
      "width": "250px",
    }
  }
/>
`;
